frontend/server/minio-helper.ts

```diff
@@ -168,7 +168,11 @@ async function parseS3ProviderInfo(
     config.useSSL = undefined;
   } else {
     if (providerInfo.Params.endpoint) {
-      const parseEndpoint = new URL(providerInfo.Params.endpoint);
+      const url = providerInfo.Params.endpoint
+      // this is a bit of a hack to add support for endpoints without a protocol (required by WHATWG URL standard)
+      // example: <ip>:<port> format. In general should expect most endpoints to provide a protocol as serviced
+      // by the backend
+      const parseEndpoint = new URL(url.startsWith("http") ? url : `https://${url}`);
```

Comment:
This is mostly to continue supporting the old implementation for how we generate [the default minio deployments](https://github.com/kubeflow/pipelines/blob/6723d3d5a92181b7ee75304671256bbe5ac18582/backend/src/v2/objectstore/config.go#L164). 

I debated whether we should update the endpoint generated here to adhere to the  WHATWG URL standard, but at some point in the near future we will have to abandon minio anyway, so we can clean this up then,

It made little sense to try to address this here and risk whilst also maintaining backwards compatibilty.